java/README.md

```diff
@@ -0,0 +1,19 @@
+# Generate JSON cross reference files for mozsearch
```

Comment:
nit: I'd call these analysis files.  While it sounds like the underlying mechanism does have to do some cross-referencing of its own, I think the mozsearch terminology is basically that indexers produce analysis files.  (And then `crossref.rs` performs the cross-referencing.)